WIP Additional Modules for Achilles

uaon opened the pull request:
When merged this pull request will: Add 2 modules for Achilles
+ Power Outage module, this module allows the option of damaging all lightpoles / lamps within a radius to give the effect of a power outage in a city or otherwise.

+ Timer Countdown module, this gives the option of having a countdown without the mission ending as the vanilla module will end the mission when counter reaches 0

- Removed Tab indents from last pull request.
+ Added entries into fn_initModules.sqf and CfgFunctions.hpp for both modules to be added.

> Unsure of proper placement for modules in the Achilles tree. feel free to make changes as necessary to any / all parts of the modules. 

> Params seemed to consistently give errors when attempted with the Power outage module, Defined variables by type instead of inside a parameter. further revision might be necessary.

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf:
```diff
@@ -0,0 +1,51 @@
+
+  private _OutageRadius = _dialogResult select 0;
+  private _OutageRadius = parseNumber _OutageRadius;
+  Private _OutageArray = [
```
Possible add all cup lamps as well?

uaon replied:
I'll check through Cups terrains / ..mod and see if I can find a class list for those to add into the array. Also from further testing it appears Fuel stations are exempt. still working out what is needed for those as well.

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf:
```diff
@@ -0,0 +1,51 @@
+private _OutageArray = [];
+private _OutageRadius = [""];
+
+    _dialogResult = [
```
weird indent going on here.

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf:
```diff
@@ -0,0 +1,54 @@
+
+private _Ctimer = _dialogResult select 0;
+
+        _time = [] call bis_fnc_countdown;
```
missing private as well as weird indent

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf:
```diff
@@ -0,0 +1,54 @@
+if (count _dialogResult == 0) exitWith {};
+
+
+private _Ctimer = _dialogResult select 0;
```
is already private due to params.

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf:
```diff
@@ -0,0 +1,54 @@
+
+        _time = [] call bis_fnc_countdown;
+        if (_time <= 0) then {_time = estimatedendservertime - servertime;};
+        _value = _time call compile _Ctimer;
```
missing private

uaon commented:
@ColdEvul Made changes to files as per your provided guidance,
+ Added ability to switch power on/off lights using `BIS_fnc_switchLamp;` to PowerOutage module
+ Set private variables in params or used `private`
+ removed several excessive indents, should appear more pleasing.

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf:
```diff
@@ -0,0 +1,88 @@
+ *
+ * Public: No
+ */
+[[], [""], []] call {
```
what is that??? O.o

ColdEvul replied:
this is enough;
```suggestion
params ["_OutageRadius", "_OutageArray", "_center"];
```

ColdEvul replied:
you can also do like this:
```suggestion
params [
    ["_OutageRadius", 0, [0]], 
    ["_OutageArray", 0, [0]],
    ["_center", 0, [0]]
];
```
To add a **default** and **expected value**.
0 = number
"" = for string

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf:
```diff
@@ -0,0 +1,52 @@
+} else {
+    _value call bis_fnc_countdown;
+    };
+} else {
```
this is weirdly set up and will cause errors

ColdEvul replied:
this in particular is not due to indent its due to the setup of the if sats.
For default structure of if sats here for used here is the following format:
```
if (_condition) then {
    //Code
} else {
    //Code
};
if (_condition) then { /*code*/};
```

But above you have done like this witch make no sence:
```
if (ismultiplayer) then {
    estimatedtimeleft _value;
} else {
    _value call bis_fnc_countdown;
    };
} else {
if (ismultiplayer) then {
    estimatedtimeleft 1e10;
} else {
    -1 call bis_fnc_countdown;
};
```
line 42
starts with `}; } else {`

ColdEvul replied:
think I see what happend here now it's the indent. I run like this:
```
if () then {
    if () then {
        if () then {
        } 
    } else {
    };
} ;

```

ColdEvul commented on cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf:
```diff
@@ -0,0 +1,88 @@
+    "Land_Lampa_Ind_EP1"
+];
+{
+[_x, _pstate] call BIS_fnc_switchLamp;
```
slap a indent here for readability

ColdEvul replied:
Like this:
```suggestion
{
    //_x setHit ["light_1_hitpoint", 0.97]; 
    //_x setHit ["light_2_hitpoint", 0.97]; 
    //_x setHit ["light_3_hitpoint", 0.97]; 
    //_x setHit ["light_4_hitpoint", 0.97];
} forEach nearestObjects [_center, _OutageArray, _OutageRadius];
```
